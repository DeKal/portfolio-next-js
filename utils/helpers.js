import { INIT_SELECTED_PAGE } from '~/consts/initState'
import $ from 'jquery'

export const initHashLocation = () => {
  if (location.hash) {
    const offset = $(location.hash).offset()
    window.scrollTo(offset.left, offset.top)
  }
}

export const isSafari = () => {
  var userAgent = navigator.userAgent.toLowerCase()
  if (userAgent.indexOf('safari') != -1) {
    // Safari agent will not have Chrome
    if (userAgent.indexOf('chrome') > -1) {
      return false
    } else {
      return true
    }
  }
}

export const setJqueryScrollEvent = () => {
  $('a').on('click', function(event) {
    // Make sure this.hash has a value before overriding default behavior
    if (this.hash) {
      // Prevent default anchor click behavior
      event.preventDefault()
      // Store hash
      var hash = this.hash

      $('html, body').animate(
        {
          scrollTop: $(hash).offset().top
        },
        400,
        function() {
          // Add hash (#) to URL when done scrolling (default click behavior)
          window.location.hash = hash
        }
      )
    } // End if
  })
}

export const cleanUpScrollEvent = () => {
  $('a').unbind()
}

export const getSelectedSection = () => {
  if (location.hash.length !== 0) {
    const selectedPage = location.hash.substr(1)
    return selectedPage
  }
  return INIT_SELECTED_PAGE
}

export const isChristmasDay = () => {
  const today = new Date()
  const month = today.getMonth() + 1
  const isDecember = month == 12
  const day = today.getDate()
  const isChristMasTime = 20 <= day <= 30

  return isDecember && isChristMasTime
}

export const isNewYearComing = () => {
  const today = new Date()
  const day = today.getDate()
  const isFirstTwoWeek = 1 <= day <= 16
  const month = today.getMonth() + 1
  const isJanuary = month == 1

  return isJanuary && isFirstTwoWeek
}
