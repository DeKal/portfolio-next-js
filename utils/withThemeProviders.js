import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import defaultTheme from '~/views/theme'

export const shallowWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

export const mountWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}
