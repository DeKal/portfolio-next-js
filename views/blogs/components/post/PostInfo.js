import React from 'react'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

const PostInfo = ({ authorName, date, minRead, avatar }) => {
  return (
    <InfoRow>
      <Col md={6}>
        <Row>
          <ColNoGutter xs={'auto'} sm={'auto'}>
            <Avatar alt={authorName} src={avatar} />
          </ColNoGutter>
          <Info>
            <Row noGutters>
              <Author> {authorName} </Author>
            </Row>
            <Row noGutters>
              <SubInfo>{`${date} - ${minRead} mins read`}</SubInfo>
            </Row>
          </Info>
        </Row>
      </Col>
      <Col md={{ span: 2, offset: 4 }}></Col>
    </InfoRow>
  )
}

PostInfo.propTypes = {
  authorName: PropTypes.string,
  date: PropTypes.string,
  minRead: PropTypes.string,
  avatar: PropTypes.string
}

export default PostInfo

const InfoRow = styled(Row)`
  margin-top: 32px;
`
const Avatar = styled.img`
  width: 48px;
  height: 48px;
  border-radius: 50%;
  display: block;
  vertical-align: middle;
`
const Info = styled(Col)`
  display: block;
  width: 100%;
  padding-left: 12px;
`
const SubInfo = styled.div`
  font-weight: 300;
  font-size: 14px;
  color: rgba(0, 0, 0, 0.54);
`

const Author = styled.div`
  line-height: 20px;
  font-size: 16px;
  color: rgba(0, 0, 0, 0.84);
  font-weight: 400;
  margin-bottom: 2px;
`
const ColNoGutter = styled(Col)`
  padding-left: 0;
  padding-right: 0;
`
