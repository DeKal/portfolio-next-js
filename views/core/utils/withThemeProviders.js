import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import homeTheme from '~/theme/theme'
import baseTheme from '~/theme/baseTheme'

/* istanbul ignore next */
export const shallowWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

/* istanbul ignore next */
export const shallowWithDivingTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children).dive()
}

/* istanbul ignore next */
export const mountWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}

/* istanbul ignore next */
export const shallowWithBaseTheme = (children, theme = baseTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}
