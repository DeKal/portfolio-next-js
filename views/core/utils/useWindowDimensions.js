import { useState, useEffect } from 'react'
import { getWindowDimensions } from '~/views/core/utils/helpers'

/* istanbul ignore next */
const useWindowDimensions = () => {
  const [windowDimensions, setWindowDimensions] = useState(
    getWindowDimensions()
  )

  useEffect(() => {
    /* istanbul ignore next */
    function handleResize() {
      setWindowDimensions(getWindowDimensions())
    }

    window.addEventListener('resize', handleResize)
    /* istanbul ignore next */
    return () => window.removeEventListener('resize', handleResize)
  }, [])

  return windowDimensions
}

export default useWindowDimensions
