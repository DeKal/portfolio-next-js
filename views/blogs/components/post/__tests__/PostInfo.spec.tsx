import * as React from 'react'
import { shallowWithTheme } from '~/utils/withThemeProviders'
import PostInfo from '~/views/blogs/components/post/PostInfo'

describe('Blog PostInfo', () => {
  it('should render without throwing an error', function () {
    const wrap = shallowWithTheme(<PostInfo />)
    expect(wrap).toMatchSnapshot()
  })
})
