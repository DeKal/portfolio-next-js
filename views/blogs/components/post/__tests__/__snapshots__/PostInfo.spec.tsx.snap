// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog PostInfo should render without throwing an error 1`] = `
<PostInfo__InfoRow>
  <Col
    md={6}
  >
    <ForwardRef
      noGutters={false}
    >
      <PostInfo__ColNoGutter
        sm="auto"
        xs="auto"
      >
        <PostInfo__Avatar />
      </PostInfo__ColNoGutter>
      <PostInfo__Info>
        <ForwardRef
          noGutters={true}
        >
          <PostInfo__Author>
             
             
          </PostInfo__Author>
        </ForwardRef>
        <ForwardRef
          noGutters={true}
        >
          <PostInfo__SubInfo>
            undefined - undefined mins read
          </PostInfo__SubInfo>
        </ForwardRef>
      </PostInfo__Info>
    </ForwardRef>
  </Col>
  <Col
    md={
      Object {
        "offset": 4,
        "span": 2,
      }
    }
  />
</PostInfo__InfoRow>
`;
